### ISSUE TITLE
Implementation detail request for classification task

### ISSUE THREAD

**Issue by changliu816** [yzou2/CRST#23]
Hi,

For implementation detail for Office-31 and Visda17, do you also follow those two steps?
1) Pretraining on the source dataset
2) Self-training on the target dataset

If so, could you please provide how many epochs you train for source dataset in step 1) for two datasets and how many epochs you train for step 2) ? 

Best,
Chang

**Comment by yzou2** [comment]
Yes, I pretrained on source and then self-train on target. For visda17, I just train 3 epochs on source and 20 epochs on target. For Office-31, pretrain for 200 epochs and self-traing for 50 epochs.

**Comment by changliu816** [comment]
Hi Dr.Zou,

There is one more thing confusing me. In segmentation, you define each step1) and step 2) as one round and then you have 4 round, and in each round, you train 2 epochs.

In your description above e.g Office-31, you self-train model for 50 epochs at step 2).  May I know **how many rounds** do you train and **how many epochs in each round**?   

Thanks,
Chang

**Comment by yzou2** [comment]
Since pseudo-label generation is time-costly for semantic segmentation, we generate pseudo-label for segmentation every two epochs. However, for pseudo-label generation for image classification, it's much faster. Thus we generate pseudo-label every epoch for classification.
